### ISSUE TITLE
compute_pairwise_loss() in the code

### ISSUE THREAD

**Issue by wxdrizzle** [MIDA-group/CoMIR#51]
Hello, and thank you so much for your work! The CoMIR does enlighten me a lot. I appreciate your time so I'm trying to make my question short.

I just have a question about the `compute_pairwise_loss()` function in `train-biodata.ipynb`. I noticed that you are using `softmaxes[i] = -pos + torch.logsumexp(neg, dim=0)` to compute the loss. If my understanding is correct, this corresponds to calculate 

<img src="https://user-images.githubusercontent.com/46703360/139607029-b88637be-4e56-431b-88e5-c8b727c7b0c1.png" width=30%>

But the InfoNCE loss mentioned in your paper is 
<img src="https://user-images.githubusercontent.com/46703360/139607168-fa323fc9-a02e-4221-a6c2-659b92291a8d.png" width=40%>
which contains the similarity of the positive pair in the denominator. 

Although there is only some slight difference between the two formulas, I'm not sure if it will lead to change of training performance. So, could you please clarify whether you are using the first formula, and why?

**Comment by npielawski** [comment]
Hello,

Sorry for the delay answering. I looked into it and what you mention seems correct: it looks wrong in the code. The following computations and results might still be correct however; the notebook on Github is a cleaned version of the original code. I will look further into it whenever I can.

Merry Christmas! 🎅 

Best,
Nicolas

**Comment by wxdrizzle** [comment]
Hi Nicolas,

Thank you for the clarification! Based on your reply, I think I may just need to use the second (original) version of this loss. I'll close this issue, but if you have time on this in the future, I will still be very happy to get some further discussions in this issue. Again, thanks for your wonderful work!

Merry Christmas!
Sincerely,
Xin

**Comment by BailiangJ** [comment]
```
    # Computation of the loss, one row after the other.
    irange = np.arange(2*N)
    softmaxes = torch.empty(2*N).to(device)
    for i in range(2*N):
        j = (i + N) % (2 * N)
        pos = similarities[i, j]
        # The negative examples are all the remaining points
        # excluding self-similarity
        neg = similarities[i][irange != i]
        softmaxes[i] = -pos + torch.logsumexp(neg, dim=0)
    return softmaxes, similarities
```
Hi @wxdrizzle , I think the code is correct here, the loss computation is consistent with InfoNCE. 

The name variable `neg`  is misleading, it is the set of all negative samples and **one positive samples**， since it's only excluding self-similarity.
